uni/lib/view/course_unit_info/widgets/course_unit_classes.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/controller/load_info.dart';
+import 'package:uni/model/entities/course_units/course_unit_class.dart';
+import 'package:uni/model/entities/session.dart';
+import 'package:uni/model/providers/session_provider.dart';
+import 'package:uni/view/course_unit_info/widgets/course_unit_info_card.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class CourseUnitsClassesView extends StatelessWidget {
+  final List<CourseUnitClass> classes;
+  const CourseUnitsClassesView(this.classes, {super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final Session session = context.read<SessionProvider>().session;
+    final List<CourseUnitInfoCard> cards = [];
+    for (var courseUnitClass in classes) {
+      final bool isMyClass = courseUnitClass.students
+          .where((student) =>
+              student.number ==
+              (int.tryParse(
+                      session.studentNumber.replaceAll(RegExp(r"\D"), "")) ??
+                  0))
+          .isNotEmpty;
+      cards.add(_buildCard(
+          isMyClass
+              ? '${courseUnitClass.className} *'
+              : courseUnitClass.className,
+          Column(
+            children: courseUnitClass.students
+                .map((student) => _buildStudentWidget(student, session))
+                .toList(),
+          )));
+    }
+
+    return Container(
+        padding: const EdgeInsets.only(left: 10, right: 10),
+        child: ListView(children: cards));
+  }
+
+  CourseUnitInfoCard _buildCard(String sectionTitle, Widget sectionContent) {
+    return CourseUnitInfoCard(
+      sectionTitle,
+      sectionContent,
+    );
+  }
+
+  Widget _buildStudentWidget(CourseUnitStudent student, Session session) {
```

Comment:
Can you remove this helper function, and create a class StudentRow or StudentWidget?